Add osu!lazer tournament client article

MiraiSubject opened the pull request:
This adds instructions on how to setup the osu!lazer tournament client, the very barebones of just making it work. 

There's a lot to cover about this client, so I believe making smaller PRs is probably a better idea to keep it reviewable.

paper-toaster commented:
`/setup` should prob be a section on `/osu!lazer_tournament_client` instead of its own article.

This would prob be classified as a stub

MiraiSubject commented:
Alright, changed it to a single file now and added the stub marker as well.

MiraiSubject commented:
I've added information about customising the client with flags, mods and videos. Wording of the sentences can definitely be improved, because some of them still feel a bit awkward to read or are just awkwardly placed in general.

paper-toaster commented:
Would this article be exempt from formal writing? Because it's like half-guide and half-regular article

MiraiSubject commented:
There's going to be more guide-like content in this explaining how the different editors work (and what their purpose is) and what the controls do in the different scenes with screenshots. I also plan to port over a guide on how to utilize the green chroma keying optimally in OBS (if I have the time I might also look into this for other streaming programs)

Hopefully this helps the decision process about it somewhat.

cl8n commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,66 @@
+# osu!lazer tournament client
+
+**osu!lazer tournament client** is the official client that overlays osu!tourney.
```
putting "osu!lazer tournament client" at the beginning makes it sound like a title when it's not

also the description doesn't really say anything. what does it do (u already got this a little with "overlays osu!tourney")? how does it compare to using the stable tourney client alone?

cl8n commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,66 @@
+
+**osu!lazer tournament client** is the official client that overlays osu!tourney.
+
+Users encountering problems with the client may [create an issue](https://github.com/ppy/osu/issues) or send an email to [tournaments@ppy.sh](mailto:tournaments@ppy.sh).
```
```suggestion
Users encountering problems with the client may [create an issue on GitHub](https://github.com/ppy/osu/issues) or send an email to [tournaments@ppy.sh](mailto:tournaments@ppy.sh).
```

cl8n commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,66 @@
+
+## Setup
+
+To start the osu!lazer tournament client, you need to specify a launch argument to the executable.
```
```suggestion
To start the osu!lazer tournament client, you need to specify a launch argument to the osu!lazer executable.
```

cl8n commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,66 @@
+
+To start the osu!lazer tournament client, you need to specify a launch argument to the executable.
+
+To do this, create a new shortcut on the desktop and set the location for it to `%LOCALAPPDATA%\osulazer\osu!.exe --tournament`. This will make that specific shortcut start up osu!lazer in the tournament client mode.
```
can be part of the previous paragraph

cl8n replied:
seems like an odd way to introduce making the shortcut, it's written as if this is the only way to "specify a launch argument to the executable"

cl8n commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,66 @@
+
+To do this, create a new shortcut on the desktop and set the location for it to `%LOCALAPPDATA%\osulazer\osu!.exe --tournament`. This will make that specific shortcut start up osu!lazer in the tournament client mode.
+
+Because the osu!lazer tournament client is only an overlay for osu!tourney, it will need to be set up as well. To make sure osu!tourney will work with osu!lazer you will need to set the Release stream in osu!tourney to Cutting Edge (Experimental) and create an empty file called `ipc.txt` inside the installation folder of osu!tourney. After that follow the [osu!tourney setup](/wiki/osu!tourney/Set ...
```
```suggestion
Because the osu!lazer tournament client is only an overlay for osu!tourney, that will need to be set up as well. Set the Release stream in osu!tourney to `Cutting Edge (Experimental)` and create an empty file called `ipc.txt` inside the installation folder of osu!tourney. After that, follow the [osu!tourney setup guide](/wiki/osu!tourney/Setup).
```

cl8n commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,66 @@
+
+Because the osu!lazer tournament client is only an overlay for osu!tourney, it will need to be set up as well. To make sure osu!tourney will work with osu!lazer you will need to set the Release stream in osu!tourney to Cutting Edge (Experimental) and create an empty file called `ipc.txt` inside the installation folder of osu!tourney. After that follow the [osu!tourney setup](/wiki/osu!tourney/Set ...
+
+Open the osu!lazer tournament client using the shortcut and you will see this setup screen.
```
```suggestion
Open the osu!lazer tournament client and you will see this setup screen:
```

cl8n commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,66 @@
+
+![osu!lazer tournament client setup screen](img/setup-screen.png)
+
+- Make sure the Current IPC Source matches the location of the osu!tourney instance that you will be using.
```
```suggestion
- Make sure the `Current IPC source` matches the location of the osu!tourney instance that you will be using.
```

cl8n commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,66 @@
+![osu!lazer tournament client setup screen](img/setup-screen.png)
+
+- Make sure the Current IPC Source matches the location of the osu!tourney instance that you will be using.
+- Login to osu!lazer by clicking on Change login.
```
```suggestion
- Log in to osu!lazer by clicking on `Change Login`.
```
I think "log in" is 2 words when you use it as a verb, could be wrong tho

cl8n commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,66 @@
+- Make sure the Current IPC Source matches the location of the osu!tourney instance that you will be using.
+- Login to osu!lazer by clicking on Change login.
+- Set the correct ruleset using the dropdown menu.
+- Change the height to match the `Height` you had set in the `tournament.cfg` file of osu!tourney.
```
```suggestion
- Change the height to match the `Height` set in the `tournament.cfg` file of osu!tourney.
```

cl8n commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,66 @@
+
+## Customisation
+
+The osu!lazer tournament client can be customised by providing it with custom flags, mod icons and video files. These will get displayed in the respective scenes as needed.
```
```suggestion
The osu!lazer tournament client can be customised by providing it with custom flags, mod icons, and video files. These will display in the respective scenes as needed.
```

cl8n commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,66 @@
+
+### Flags
+
+For custom flags `.jpg` and `.png` files are accepted. These flags have to be at a resolution of 140x94 to be displayed correctly.
```
```suggestion
For custom flags, `.jpg` and `.png` files are accepted. These flags have to be at a resolution of 140x94 to be displayed correctly.
```

cl8n replied:
might want to explain here why you'd want custom flags in the first place, it's probably not obvious that these can refer to teams and not necessarily countries

cl8n commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,66 @@
+
+For custom flags `.jpg` and `.png` files are accepted. These flags have to be at a resolution of 140x94 to be displayed correctly.
+
+An example of a flag using the correct specifications is this flag of Australia:
+
+![][flag_AU]
```
this isn't using correct spec though. example might not be important for a simple image anyway

cl8n commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,66 @@
+
+### Mods
+
+For custom mod icons `.jpg` and `.png` files are accepted. The resolution can be anything and the client will try to auto fit it in the beatmap panel. For reference, a beatmap panel at 1920x1080 is 563x60 pixels.
```
```suggestion
For custom mod icons, `.jpg` and `.png` files are accepted. The resolution can be anything and the client will fit it in the beatmap panel. For reference, a beatmap panel at 1920x1080 is 563x60 pixels.
```
what's the ideal mod resolution then? 60x60? u should state that, instead of the map panel res

cl8n commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,66 @@
+
+### Videos
+
+Video files in the background of scenes visible to the stream can be displayed. These will automatically be looped in the client once the end of the file is reached.
```
```suggestion
Videos can be displayed in the backgrounds of scenes visible to the stream. These will automatically be looped in the client once the end of the video is reached.
```

cl8n commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,66 @@
+The files have to adhere to the following specifications:
+
+- 16:9 aspect ratio, for example 1280x720 or 1920x1080
+- The file extension is `mp4`, `m4v`, or `avi`.
```
```suggestion
- `mp4`, `m4v`, or `avi` file extension
```
to be consistent with the aspect ratio line

cl8n replied:
should specify what video and audio codecs lazer supports

cl8n commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,66 @@
+| Seeding | `seeding` |
+| MapPool | `mappool` |
+| Gameplay | `gameplay` |
+| Win | `teamwin-red` (if team red is the winner) / `teamwin-blue` (if team blue is the winner) |
```
i dont think it's right to say "if" in these, cuz you'd always want both videos. the naming is so self-explanatory i think you could do without the comments too
```suggestion
| Win | `teamwin-red`, `teamwin-blue` |
```

peppy commented:
Note that configuration and resource locations have also [changed recently](https://github.com/ppy/osu/pull/9256) to allow multiple configurations to coexist.

peppy commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -37,18 +37,18 @@ Flags have to be placed in `%APPDATA%\osu\tournament\flags`. The flags can then
-- The file extension is `mp4`, `m4v`, or `avi`.
+- `mp4`, `m4v`, or `avi` file extension
 
 The video files have to placed in `%APPDATA%\osu\tournament\videos` and specific names are required for the correct functionality.
```
as mentioned, these paths are now wrong

peppy commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -37,18 +37,18 @@ Flags have to be placed in `%APPDATA%\osu\tournament\flags`. The flags can then
-For custom mod icons `.jpg` and `.png` files are accepted. The resolution can be anything and the client will try to auto fit it in the beatmap panel. For reference, a beatmap panel at 1920x1080 is 563x60 pixels.
+For custom mod icons, `.jpg` and `.png` files are accepted. The resolution can be anything and the client will fit it in the beatmap panel. For reference, a beatmap panel at 1920x1080 is 563x60 pixels.
 
 Mod icons have to be placed in `%APPDATA%\osu\tournament\mods`. The mods can then be referenced by their filenames without the file extension in the Rounds Editor and in the Seeding Results Editor.
```
needs to be updated with some supplementary documentation

Joehuu reviewed (changes requested):
Will wait for the reviews to be resolved, but the image needs to be updated as per ^ (compile and build as of now).

MiraiSubject commented:
> Will wait for the reviews to be resolved, but the image and string references need to be updated as per ^ (compile and build as of now).

I'll try to get to it asap once I'm settled in the new place. Apologies for taking a while to resolve these.

TicClick commented:
> > Will wait for the reviews to be resolved, but the image and string references need to be updated as per ^ (compile and build as of now).
> 
> I'll try to get to it asap once I'm settled in the new place. Apologies for taking a while to resolve these.

@MiraiSubject so, how did it go?

MiraiSubject commented:
@TicClick I totally forgot about this PR. I was supposed to get on it after I got https://github.com/ppy/osu/pull/11118 merged, so I had an easier time explaining how to switch between tournaments in the new situation without fiddling with the `ini` file.

MiraiSubject commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,87 @@
+
+![][flag_AU]
+
+Flags have to be placed in `<your-tournament>\flags`. The flags can then be referenced in the Team Editor by their filenames without the file extension.
```
Not really sure how to handle these. I called it `your-tournament` in the example directory tree too, but I don't know if there's a way to make this more clear.

MiraiSubject commented:
I have addressed most of the reviews for now. 

For the beatmap panel one I need to check with the code and visually again and for the description I'm just not sure if the one I added is adequate enough.

peppy commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,86 @@
+
+### Flags
+
+For custom flags, `.jpg` and `.png` files are accepted. These flags have to be at a resolution of 140x94 to be displayed correctly. It will be scaled to fill the shape by the client if the resolution does not match.
```
What about default flags? Nowhere is it said what is supported for defaults.

peppy commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,86 @@
+
+### Flags
+
+For custom flags, `.jpg` and `.png` files are accepted. Flag images should be at least 140x94, maintaining an aspect ratio close to this specification for best results.
```
Nowhere are the default flags mentioned (ie. what is supported).

peppy commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,86 @@
+
+### Mods
+
+For custom mod icons, `.jpg` and `.png` files are accepted. The resolution can be anything and the client will fit it in the beatmap panel. For reference, a beatmap panel at 1920x1080 is 563x60 pixels.
```
Are there any non-"custom" mod icons? I don't think there are, which means these are not custom but *required*. If not, the defaults should probably be linked somewhere?

peppy commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,86 @@
+
+### Videos
+
+Videos can be displayed in the backgrounds of scenes visible to the stream. These will automatically be looped in the client once the end of the video is reached.
```
```suggestion
Looping videos can be displayed in the background of each scene.
```

peppy commented on wiki/osu!tourney/osu!lazer_tournament_client/en.md:
```diff
@@ -0,0 +1,86 @@
+
+The osu!lazer tournament client has three editors that can be used to provide the following information about a tournament: the teams and its players, the rounds and their accompanying mappool and the bracket. 
+
+### Team Editor
```
Is this document supposed to end here? Should this not be added as a separate step if you don't intend on finishing it?

peppy commented:
This whole thing feels really half-assed effort. I'd recommend fixing the remaining issues and getting it merged so more people can contribute efficiently.

MiraiSubject commented:
Removed the references to the editor (that was a sloppy commit on my side). Other than that addressed the review apart from the built-in mods which depends on https://github.com/ppy/osu/pull/11589.

I got a change for the mods on wiki side ready locally, but don't want to create a mess of commits if any more design changes are required on the game side PR.

peppy commented:
Let's go with this and build off it in smaller pieces.
